extensions/ql-vscode/src/data-extensions-editor/extension-packs.ts

```diff
@@ -0,0 +1,92 @@
+import { relative } from "path";
+import { window } from "vscode";
+import { CodeQLCliServer } from "../cli";
+import { getOnDiskWorkspaceFolders, showAndLogErrorMessage } from "../helpers";
+import { ProgressCallback } from "../progress";
+
+export async function pickExtensionPack(
+  cliServer: CodeQLCliServer,
+  progress: ProgressCallback,
+): Promise<string | undefined> {
+  progress({
+    message: "Resolving extension packs...",
+    step: 1,
+    maxStep: 3,
+  });
+
+  // Get all existing extension packs in the workspace
+  const additionalPacks = getOnDiskWorkspaceFolders();
+  const extensionPacks = await cliServer.resolveQlpacks(additionalPacks, true);
+  const options = Object.keys(extensionPacks).map((pack) => ({
+    label: pack,
+    extensionPack: pack,
+  }));
+
+  progress({
+    message: "Choosing extension pack...",
+    step: 2,
+    maxStep: 3,
+  });
+
+  const extensionPackOption = await window.showQuickPick(options, {
+    title: "Select extension pack to use",
+  });
+  if (!extensionPackOption) {
+    return undefined;
+  }
+
+  const extensionPackPaths = extensionPacks[extensionPackOption.extensionPack];
+  if (extensionPackPaths.length !== 1) {
+    void showAndLogErrorMessage(
+      `Extension pack ${extensionPackOption.extensionPack} could not be resolved to a single location`,
+    );
+    return undefined;
+  }
+
+  return extensionPackPaths[0];
+}
+
+export async function pickModelFile(
+  cliServer: CodeQLCliServer,
+  progress: ProgressCallback,
+  extensionPackPath: string,
+): Promise<string | undefined> {
+  // Find the existing model files in the extension pack
+  const additionalPacks = getOnDiskWorkspaceFolders();
+  const extensions = await cliServer.resolveExtensions(
+    extensionPackPath,
+    additionalPacks,
+  );
+
+  const modelFiles = new Set<string>();
+
+  if (extensionPackPath in extensions.data) {
```

Comment:
What happens if this if statement is false? This is a legitimate case if the qlpack does not have any data extensions in it (but it might have some ml-model extensions).

Then the array will be empty and no options will be presented in the quick picker. Is this handled by the follow up PR where you add an option to create a new file?

Maybe we can skip this modal in this case and go right to askign the user to create a new file.

Reply:
Good point, I'll add this in the follow-up PR (#2296) so that we skip this quick pick in case it's empty.

Reply:
I've added this in https://github.com/github/vscode-codeql/pull/2296/commits/59c0610122843951afba8271b8a170bf9c64f6aa.